hr_expense_invoice/models/hr_expense_sheet.py

```diff
@@ -112,3 +143,30 @@ def action_view_invoices(self):
             action["view_mode"] = "tree,form"
             action["domain"] = [("id", "in", invoice_ids)]
         return action
+
+    def _compute_state(self):
+        """Set proper state according linked invoices."""
+        sheets_with_invoices = self.filtered(
+            lambda sheet: all(
+                expense.invoice_id and expense.invoice_id.state == "posted"
+                for expense in sheet.expense_line_ids
+            )
+            and sheet.state == sheet.approval_state
+        )
+
+        company_account_sheets = sheets_with_invoices.filtered(
+            lambda sheet: sheet.payment_mode == "company_account"
+        )
+        company_account_sheets.write({"state": "done"})
+
+        sheets_with_paid_invoices = (
+            sheets_with_invoices - company_account_sheets
+        ).filtered(
+            lambda sheet: all(
```

Comment:
You could filter by previous state to discard invalid candidates, e.g. draft ones.

Reply:
This is already filtered in [line 155](https://github.com/OCA/hr-expense/pull/256/commits/1a6445ba59c057b0dd0cdc55a3d04f3f96bb3a83#diff-406b9f2af65ef4b7b24c6638a958cf7301ddad72e846bc1c68083c281238282dR154). This is the [original compute](https://github.com/odoo/odoo/blob/17.0/addons/hr_expense/models/hr_expense_sheet.py#L293).